extensions/shared/register-jetpack-block.js

```diff
@@ -48,6 +50,14 @@ export default function registerJetpackBlock( name, settings, childBlocks = [] )
 		...settings,
 		title: betaExtensions.includes( name ) ? `${ settings.title } (beta)` : settings.title,
 		edit: requiredPlan ? wrapPaidBlock( { requiredPlan } )( settings.edit ) : settings.edit,
+		icon: requiredPlan ? (
+			<>
```

Comment:
Need to bump `@automattic/calypso-build` to a more recent version for this to work.